### PR TITLE
Add support for faraday 2.x

### PR DESCRIPTION
**Pull request by lavoiesl**
Includes a rebased version of https://github.com/JsonApiClient/json_api_client/pull/402

Fixes https://github.com/JsonApiClient/json_api_client/issues/407

- Prefix activesupport appraisals, as to not conflict with faraday appraisals
- Bump the minimum Faraday requirement to 1.10 to pick up json support: https://github.com/lostisland/faraday/pull/1400
- Allow Faraday 2.x, it doesn't require any more change than using 1.10's json support
- Remove the obsolete `ParseJson` middleware
- Add Faraday 1.10 and 2.x appraisals

Notice how no tests have changed

### CODE REVIEW

**Comment by sarahsehr** [comment]
@gaorlov Bump please :) Faraday 2.0 has been out for almost 2 years.

**Comment by timbogit** [comment]
@gaorlov Is this still being considered? My team depends heavily `json_api_client` as well as `faraday`, so we would really appreciate this being merged. 🙇

**Comment by dolan** [comment]
bumping again @gaorlov same situation here as @timbogit

**Comment by gaorlov** [comment]
Hi, folks!
Sorry about the delayed response and thank you for the contributions! 
Quick question: what is the thinking behind removing gemfiles 5.2 and 6.0?

**Comment by lavoiesl** [comment]
I don't remember making a conscious decision to remove those. Looking at the commit, it looks live it was auto-removed when running the appraisal generator. 

Is it possible that those are obsolete files?

**Comment by gaorlov** [comment]
i must admit i am not  up to date on rails versions. It looks like 6.0 has reached EOL, but 6.1 is still supported. Can we bring it back and just test against `6` instead of a specific sub-version? i don't have a strong opinion on 5.2.3

**Comment by lavoiesl** [comment]
Looking at the diff again, it does seem like an outdated file. It doesn't actually require rails, just activesupport

**Comment by gaorlov** [comment]
> Looking at the diff again, it does seem like an outdated file. It doesn't actually require rails, just activesupport

is there another mechanism by which we test that activesupport version?

also: looks like there's a bunch of old ruby failures. i think it's probably okay to drop ruby 2.3, 2.4, and 2.5 from the test matrix if you want to drop them from the workflow

**Comment by lavoiesl** [comment]
New matrix:
- Ruby: 3.0, 3.1, 3.2, 3.3
- Activesupport: 6.0, 6.1, 7.0, 7.1
- Faraday: 1.x, 2.x

Updated the changelog as well

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -22,27 +22,11 @@ jobs:
...
-          - ruby: "2.6"
-            rails: "4.1"
-          - ruby: "2.6"
-            rails: "4.2"
-          - ruby: "2.7"
-            rails: "4.1"
-          - ruby: "2.7"
-            rails: "4.2"
+        ruby:
+          - "3.0"
```
Comment (lavoiesl) [inline .github/workflows/ruby.yml:26]: Needs to be quoted, otherwise `3.0` becomes `3`, which gets interpreted as `3.x`, resolving to `3.3`

**Comment by gaorlov** [comment]
marvelous. thank you for pulling in `appraisals`! that's a really nice tool

**Comment by lavoiesl** [comment]
Oddly enough, it was there as a dependency and the configurations were there, but it wasn't being used in the tests 🤷🏼‍♂️

**Comment by gaorlov** [comment]
thank you for you contributions and patience!
[1.23.0](https://rubygems.org/gems/json_api_client/versions/1.23.0) is now live with your changes

**Pull request merged**